providers/gce/gce_instances.go

```diff
@@ -92,6 +93,58 @@ func (g *Cloud) ToInstanceReferences(zone string, instanceNames []string) (refs
 	return refs
 }
 
+// cmpAddressFamily compares IPv6 and IPv4 address families so that IPv6 is smaller than IPv4.
+func cmpAddressFamily(a, b string) int {
+	// IPv6 addresses contain a colon („:”), IPv4 addresses do not.
+	if strings.Contains(a, ":") && !strings.Contains(b, ":") {
+		return -1
+	} else if !strings.Contains(a, ":") && strings.Contains(b, ":") {
+		return 1
+	}
+
+	return 0
+}
+
+// cmpAddressType compares IP address types so that internal address is smaller than external.
+func cmpAddressType(a, b v1.NodeAddressType) int {
```

Comment:
I believe this is a correct implementation right now but it depends on implementation detail of other code fragments (the original order of addresses). This is against the clear architecture rule. You should respect the original order of IPs no matter how it is implemented originally (that's what Antionio suggest to do in his solution). 

If the implementation for addresses change (i.e. order change, new node address type is added) this code fragment may be a cause of problems and bugs.

Reply:
Thanks for the explanation. I removed the part sorting by address type.